akka-docs/rst/scala/stream/stream-dynamic.rst

```diff
@@ -61,3 +61,80 @@ Refer to the below for usage examples.
+
+The resulting :class:`Source` can be materialized any number of times, each materialization effectively attaching
+a new subscriber. If there are no subscribers attached to this hub then it will not drop any elements but instead
+backpressure the upstream producer until subscribers arrive. This behavior can be tweaked by using the combinators
+``.buffer`` for example with a drop strategy, or just attaching a subscriber that drops all messages. If there
+are no other subscribers, this will ensure that the producer is kept drained (dropping all elements) and once a new
+subscriber arrives it will adaptively slow down, ensuring no more messages are dropped.
+
+Combining dynamic stages to build a simple Publish-Subscribe service
+^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
+
+The features provided by the Hub implementations are limited by default. This is by design, as various combinations
+can be used to express additional features like unsubscribing producers or consumers externally. We show here
+an example that builds a :class:`Flow` representing a publish-subscribe channel. The input of the :class:`Flow` is
+published to all subscribers while the output streams all the elements published.
+
+First, we connect a :class:`MergeHub` and a :class:`BroadcastHub` together to form a publish-subscribe channel. Once
+we materialize this small stream, we get back a pair of :class:`Source` and :class:`Sink` that together define
+the publish and subscribe sides of our channel.
+
+.. includecode:: ../code/docs/stream/HubsDocSpec.scala#pub-sub-1
+
+We now use a few tricks to add more features. First of all, we attach a ``Sink.ignore``
+at the broadcast side of the channel to keep it drained when there are no subscribers. If this behavior is not the
+desired one this line can be simply dropped.
+
+.. includecode:: ../code/docs/stream/HubsDocSpec.scala#pub-sub-2
+
+We now wrap the :class:`Sink` and :class:`Source` in a :class:`Flow` using ``Flow.fromSinkAndSource``. This bundles
+up the two sides of the channel into one and forces users of it to always define a publisher and subscriber side
+(even if the subscriber side is just dropping). It also allows us to very simply attach a :class:`KillSwitch` as
+a :class:`BidiStage` which in turn makes it possible to close both the original :class:`Sink` and :class:`Source` at the
+same time.
+Finally, we add ``backpressureTimeout`` on the consumer side to ensure that subscribers that block the channel for more
+than 3.seconds are forcefully removed.
+
+.. includecode:: ../code/docs/stream/HubsDocSpec.scala#pub-sub-3
+
+The resulting Flow now has a type of ``Flow[String, String, UniqueKillSwitch]`` representing a publish-subscribe
+channel which can be used any number of times to attach new producers or consumers. In addition, it materializes
+to a :class:`UniqueKillSwitch` that can be used to deregister a single user externally:
```

Comment:
I would love to see one last snippet that triggers the killswitch here, so I don't have to go read about it if I'm just doing something simple/playing around